src/checkqueue.h

```diff
@@ -167,16 +167,26 @@ class CCheckQueue
     //! Add a batch of checks to the queue
     void Add(std::vector<T>& vChecks)
     {
-        LOCK(m_mutex);
-        for (T& check : vChecks) {
-            queue.push_back(T());
-            check.swap(queue.back());
+        if (vChecks.empty()) {
```

Comment:
The code below assumes `vChecks` may be modified by another thread while this is executing (it cannot actually). I guess either remove that assumption or move this access to `vChecks` under `m_mutex` (if that mutex it supposed to protect `vChecks`!?).

It is a race to call `std::vector::empty()` while modifying it concurrently.

Reply:
You are right.

> I guess either remove that assumption or move this access to `vChecks` under `m_mutex` (if that mutex it supposed to protect `vChecks`!?).

The former will go.